src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java

```diff
@@ -97,6 +104,9 @@
             return castOrNull(key,getExecution());
         } else if (FlowNode.class.isAssignableFrom(key)) {
             return castOrNull(key, getNode());
+        } else if (key == EnvironmentExpander.class) {
+            // Only called when `value == null` so that it does not override expanders contributed by steps.
+            return key.cast(new PasswordParameterEnvironmentExpander(get(Run.class)));
```

Comment:
For now we just have single default implementation since I cannot think of any other similar cases to password parameters. If there are similar cases, maybe we should create an extension point for marking sensitive variables that are active from the start of the build or something.